internal/coreinternal/golden/normalize_timestamps.go

```diff
@@ -0,0 +1,198 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package golden // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/golden"
+
+import (
+	"sort"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil"
+)
+
+func normalizeTimestamps(metrics pmetric.Metrics) {
+	zeroTime := time.Unix(0, 0)
+	nanosec := pcommon.NewTimestampFromTime(zeroTime)
+	rms := metrics.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		for j := 0; j < rms.At(i).ScopeMetrics().Len(); j++ {
+			for k := 0; k < rms.At(i).ScopeMetrics().At(j).Metrics().Len(); k++ {
+				m := rms.At(i).ScopeMetrics().At(j).Metrics().At(k)
+				switch m.Type() {
+				case pmetric.MetricTypeGauge:
+					for l := 0; l < m.Gauge().DataPoints().Len(); l++ {
+						m.Gauge().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+					normGaugeDataPointSlice(m.Gauge().DataPoints())
+				case pmetric.MetricTypeSum:
+					for l := 0; l < m.Sum().DataPoints().Len(); l++ {
+						m.Sum().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+					normGaugeDataPointSlice(m.Sum().DataPoints())
+				case pmetric.MetricTypeHistogram:
+					for l := 0; l < m.Histogram().DataPoints().Len(); l++ {
+						m.Histogram().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+					normHistogramDataPointSlice(m.Histogram().DataPoints())
+				case pmetric.MetricTypeExponentialHistogram:
+					for l := 0; l < m.ExponentialHistogram().DataPoints().Len(); l++ {
+						m.ExponentialHistogram().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+					normExponentialHistogramDataPointSlice(m.ExponentialHistogram().DataPoints())
+				case pmetric.MetricTypeSummary:
+					for l := 0; l < m.Summary().DataPoints().Len(); l++ {
+						m.Summary().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+					normSummaryDataPointSlice(m.Summary().DataPoints())
+				}
+			}
+		}
+	}
+}
+
+func normGaugeDataPointSlice(ndps pmetric.NumberDataPointSlice) {
```

Comment:
Since this is used for gauge and sum
```suggestion
func normNumberDataPointSlice(ndps pmetric.NumberDataPointSlice) {
```